api/api.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+package api
+
+import (
+	"context"
+)
+
+// TriggerFunction is the type of the function that is called when a job is
+// triggered.
+// Returning true will "tick" the job forward to the next scheduled time.
+// Returning false will cause the job to be re-enqueued and triggered
+// immediately.
+type TriggerFunction func(context.Context, *TriggerRequest) bool
+
+// API is the interface for interacting with the cron instance.
+type API interface {
+	// Add adds a job to the cron instance.
+	Add(ctx context.Context, name string, job *Job) error
+
+	// Get gets a job from the cron instance.
+	Get(ctx context.Context, name string) (*Job, error)
+
+	// Delete deletes a job from the cron instance.
+	Delete(ctx context.Context, name string) error
+
+	// DeletePrefixes deletes all jobs with the given prefixes from the cron
+	// instance.
+	DeletePrefixes(ctx context.Context, prefixes ...string) error
+
+	// List lists all jobs under a given job name prefix.
+	List(ctx context.Context, prefix string) (*ListResponse, error)
```

Comment:
Changes mostly lgtm, but its a bit hard to parse for the changes for the list api given the additional refactoring/nix changes. Is there a test for the api in this PR?

Reply:
My bad, sorry about that- got semi carried away with it :grimacing: 

Here are the two places where we are testing the `List` API specifically. Happy to separate the refactor and API addition if that would make review easier.

https://github.com/diagridio/go-etcd-cron/pull/43/files#diff-4d514085b47744fcdfdda631f3b5e141857e34feb020911194a877879fcf1501R209
https://github.com/diagridio/go-etcd-cron/pull/43/files#diff-efc48b37700cd7674ce96c446ce0043c8f3d55888ddba4392b897f6411e2c30dR231